### ISSUE TITLE
Website Announcements section

### ISSUE THREAD

**Issue by cesarhernandezgt** [jakartaee/jakarta.ee#880]
The Announcements section for the website can be improved to:

- Allow smooth UI standardization of announcements.
- Announcements are hosted on the same Jakarta website.
- Layout improvement and author images.
- Follows the Pull Request workflow in the https://github.com/jakartaee/jakarta.ee/ repository for future announcements.
- Take advantage of current website infrastructure.
- Can add a scalable sidebar for future ideas.

**Comment by cesarhernandezgt** [comment]
For a preview on this feature you can check https://github.com/jakartaee/jakarta.ee/pull/879 
URL: https://deploy-preview-879--jakartaee.netlify.app/announcements/

**Comment by chrisguindon** [comment]
I really like the look and feel for this update!

Is it possible to update the PR to make sure that we are pulling announcements from the newsroom API? 

We created the Eclipse Newsroom website to make it easier for staff and the community to publish news/announcements and events on our various web properties without having to create a PR.



https://newsroom.eclipse.org/api/news?&page=1&pagesize=10&parameters[publish_to]=jakarta_ee&options[orderby][created]=asc&parameters[news_type]=announcements

**Comment by dblevins** [comment]
@chrisguindon We should let the Marketing Committee discuss, but the idea was this would be the source of truth for announcements from the Working Group, it would be completely owned by the Marketing Committee and the PR process would provide a form of committee review/collaboration before publishing.

**Comment by cesarhernandezgt** [comment]
hi @chrisguindon , thank you for the comment.

By looking at https://newsroom.eclipse.org/, this is a cool third party feed aggregator similar to https://jakartablogs.ee/. 

The main feature this GitHub issue and PR has in their scope is to actually allows one to publish content (not reference to externally hosted articles) via a PR with just the addition of a Folder, _index.md, and supporting images. Check for example this file: https://github.com/jakartaee/jakarta.ee/pull/879/files#diff-5c254c06f184bd9da914c47020c9d980

**Comment by autumnfound** [comment]
@cesarhernandezgt Newsroom is more than an aggregator in this case! It was a site we made that allows for users to submit news and events to be included across multiple sites. While most news items currently on the feed are external, content can be included in a body section of the news article submission, with the URL left empty. 

We used Drupal as the base, and articles are site content. This allows for moderation of content, using Eclipse Accounts rather than access to this repository for rights. I'll defer to Chris on how the moderation works exactly, as he's the one who pulled the system together.

**Comment by dblevins** [comment]
@autumnfound We should let the Marketing Committee discuss, but as noted the goal is to provide an official channel for public statements made by the Jakarta EE Working Group to be published somewhere on https://jakarta.ee and each post has been created and reviewed by the Working Group itself.  Further, as the exclusive channel to represent official communication of the Working Group it should not not include external material like https://jakartablogs.ee and https://newsroom.eclipse.org do.  Definitely jakartablogs.ee and newsroom could aggregate this new announcements channel.

**Comment by chrisguindon** [comment]
> The main feature this GitHub issue and PR has in their scope is to actually allows one to publish content (not reference to externally hosted articles) via a PR with just the addition of a Folder, _index.md, and supporting images. Check for example this file: https://github.com/jakartaee/jakarta.ee/pull/879/files#diff-5c254c06f184bd9da914c47020c9d980

@cesarhernandezgt @dblevins  I was under the impression that the goal here was to replace how the EF Marketing Team is posting announcements on the Jakarta.ee homepage.

I now understand that we are creating a new section to allow us to publish articles on jakarta.ee.

Articles here would be submitted to the Eclipse Newsroom website but not the other way around.

+1 from me! The more content we can publish on jakarta.ee the better! This will definitely help drive more traffic to the website!

**Comment by chrisguindon** [comment]
@cesarhernandezgt I am going to assign you on this issue since you in the Jakarta-ee organization. T

his allows me to track that we have someone looking into an issue.

**Comment by dblevins** [comment]
@chrisguindon what is the process or requirements we'd need to implement to get the Working Group Announcements hosted on jakarta.ee to feed into the newsroom aggregator and then via that reach the front page that has all announcements (working group and otherwise).  Seems like at minimum we need to ensure there is an rss feed for the Working Group Announcements.  Is it just a matter of creating that feed and submitting it to newsroom in some way?  Are there specific labels we need to use, etc?

**Comment by chrisguindon** [comment]
The newsroom site is not a content aggregator, at least not at the moment.

The process right now requires someone to submit news item via https://newsroom.eclipse.org/node/add/news

Now, with this said, It's possible to automate the process but that would require some work on the Newsroom site first. 

If we decide to move forward with this, the default rss feed that Hugo provides would probably work.

**Comment by dblevins** [comment]
Thanks, @chrisguindon.  I think I'm starting to get it, but want to confirm.  Here's what I understand:

 - Newsroom can have external content, it's just the link is manually submitted via https://newsroom.eclipse.org/node/add/news for each individual post
 - If/when the PRs merged and we wanted to get a post to show up on the front-page of jakarta.ee, we would just need to manually submit the link to newsroom

Lastly,  someday we could do either or both of these:
 - expand the capabilities of Newsroom to accept trusted rss feeds so supplying links can be automated
 - add another section to the front page of jakarta.ee specifically for posts hosted on jakarta.ee

So basically our process is manual for now, but there are options to do it in a more automated fashion in the future.

**Comment by chrisguindon** [comment]
> * Newsroom can have external content, it's just the link is manually submitted via https://newsroom.eclipse.org/node/add/news for each individual post

If you submit a URL (The canonical URL for this news item. Keep empty if you wish to host the content on this website.
), the page will redirect to the canonical URL otherwise the full article is hosted on Eclipse Newsroom.

Example of a news item with a URL:
https://newsroom.eclipse.org/news/announcements/welcome-autoware-foundation-openadx

This page will redirect to https://www.autoware.org/

Example of an Article hosted on Eclipse Newsroom:
https://newsroom.eclipse.org/news/announcements/open-source-leader-eclipse-foundation-adds-record-number-new-members-2020

In this example, the author did not submit a URL and the content is hosted on Eclipse Newsroom.

>If/when the PRs merged and we wanted to get a post to show up on the front-page of jakarta.ee, we would just need to manually submit the link to newsroom

Correct! 

> Lastly, someday we could do either or both of these:
> 
> * expand the capabilities of Newsroom to accept trusted rss feeds so supplying links can be automated

If submitting news items becomes a problem that we need to solve, we can discuss automating the submission process. I think the EF would like to continue to moderate all submission because this keeps them in the loop of what is going on in our community and it's also an opportunity for them to tag a news item to be displayed on any of our other sites if they think the content is appropriate for our other web properties.

As an example, we might want to display an announcement on jakarta.ee, eclipse.org and iot.eclipse.org

> * add another section to the front page of jakarta.ee specifically for posts hosted on jakarta.ee

It's a possibility! Tthe EF Marketing Team is responsible for the content shown on our homepage, they would need to recommend this change before we could change the homepage.

> So basically our process is manual for now, but there are options to do it in a more automated fashion in the future.

Seems about right to me!

**Comment by dblevins** [comment]
> > Lastly, someday we could do either or both of these:
> > 
> > * expand the capabilities of Newsroom to accept trusted rss feeds so supplying links can be automated
> 
> If submitting news items becomes a problem that we need to solve, we can discuss automating the submission process. I think the EF would like to continue to moderate all submission because this keeps them in the loop of what is going on in our community and it's also an opportunity for them to tag a news item to be displayed on any of our other sites if they think the content is appropriate for our other web properties.

Agree.  I don't think there'll be much news so submitting manually is no real issue.  I think we're good to go.

> > * add another section to the front page of jakarta.ee specifically for posts hosted on jakarta.ee
> 
> It's a possibility! The EF Marketing Team is responsible for the content shown on our homepage, they would need to recommend this change before we could change the homepage.

I'd phrase it slightly differently.  It's the Jakarta EE Marketing Committee, under direction of the Jakarta EE Steering Committee, who is responsible for the content shown on jakarta.ee.  But absolutely correct that the EF Marketing Team has input into both of those decision-making forums and is often the one shouldered with carrying out those decisions.

Thank you so much for all the input and explanations!

**Comment by chrisguindon** [comment]
> I'd phrase it slightly differently. It's the Jakarta EE Marketing Committee, under direction of the Jakarta EE Steering Committee, who is responsible for the content shown on jakarta.ee. But absolutely correct that the EF Marketing Team has input into both of those decision-making forums and is often the one shouldered with carrying out those decisions.

This is all documented here:
https://docs.google.com/spreadsheets/d/1efy2BjsB4zfxNGIA7ESXL_bgXZKfpfHJlRMt19zzuro/edit#gid=98299565

**Comment by trulymobilenwp** [comment]
Per my comment in issue #879 
A few points related to this request.

1. As discussed during our marketing committee call this aligns with our Q2 plans around reworking key pages on Jakarta.ee and should be addressed as that work is done so the overall look and feel comes out as intended by our overall update work.
2. I recommend that any updates to the announcements page should be done in the existing https://jakarta.ee/news/announcements/ page and/or options as currently available at https://jakarta.ee/news/
3. I recommend we determine the best process to ensure the announcements are agreed to collectively by the committee
4. As far as having announcements actually hosted as part of the Jakarta.ee domain, we should address this as part of our rework of the website but we need to ensure we will have a consistent volume of announcements to justify the work and a reasonable way to implement the support of this work.

**Comment by oliviergoulet5** [comment]
@cesarhernandezgt There has not been activity since early 2021. Can we close this issue?
